aptos-move/mvhashmap/src/versioned_data.rs

```diff
@@ -361,6 +361,40 @@ impl<K: Hash + Clone + Debug + Eq, V: TransactionWrite> VersionedData<K, V> {
         }));
     }
 
+    /// Versioned write of metadata at a given resource group key (and version). Returns true
+    /// if the previously stored metadata has changed as observed by later transactions (e.g.
+    /// metadata of a deletion can never be observed by later transactions).
+    pub fn write_metadata(
```

Comment:
stupid question for the future:
- metadata always has basically empty data - empty Bytes, rigth?
- if we change above to compare by value any value that is smaller than some size - this would be covered as well, and would not need a separate function?

Reply:
Correct, I want to change to that with the hints - it might become a bit more nuanced, e.g. might need to have a similar way of not reading if it's been observed to change, even if small, as that leads to doing validations later (after re-execution as opposed to abort).